src/Analysis/Ast/Impl/Inspection/IPythonInspector.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Python.Analysis.Inspection {
+    public interface IPythonInspector {
+        Task<ModuleMemberNamesResponse> GetModuleMemberNames(string moduleName);
```

Comment:
Probably `Async` so .NET style cop won't bark. Can it handle cancellation?

Reply:
I could modify the signature to accept a token and pass it through to the JsonRpc instance, though I'm not really sure what will happen given that I implemented the JSON-RPC server by hand in Python and it's single threaded. I'll test and see.